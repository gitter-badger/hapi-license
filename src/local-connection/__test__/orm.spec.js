import { createConnection } from '../orm';

describe('create local connection', () => {
    let sequelize;
    let log;

    beforeEach(() => {
        process.env.POSTGRES_DB = 'db_test';
        process.env.POSTGRES_USER = 'db_user';
        process.env.POSTGRES_PASSWORD = 'db_password';
        process.env.POSTGRES_HOST = 'db_localhost';
        sequelize = jest.fn((db, user, pass, options) => options);
        log = {
            trace: jest.fn()
        };
    });

    it('should create a sequelize connection', () => {
        const { logger } = createConnection(sequelize, log);
        const calls = sequelize.mock.calls[0];
        logger('something');

        expect(sequelize).toHaveBeenCalled();
        expect(sequelize).toHaveBeenCalledTimes(1);
        expect(calls).toMatchSnapshot();

        expect(log.trace).toHaveBeenCalled();
        expect(log.trace).toHaveBeenCalledTimes(1);
        expect(log.trace).toHaveBeenCalledWith('something');
    });
});
