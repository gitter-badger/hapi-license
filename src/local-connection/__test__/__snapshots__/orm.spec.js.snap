// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create local connection should create a sequelize connection 1`] = `
Array [
  "db_test",
  "db_user",
  "db_password",
  Object {
    "dialect": "postgres",
    "host": "db_localhost",
    "logger": [Function],
    "operatorsAliases": undefined,
  },
]
`;
